File: occasum-core/src/main/java/pw/stamina/occasum/properties/primitives/clamp/MaxDoubleClamp.java
package pw.stamina.occasum.properties.primitives.clamp;

final class MaxDoubleClamp implements DoubleClamp {
    private final double max;

    MaxDoubleClamp(double max) {
        this.max = max;
    }

    @Override
    public double clamp(double value) {
        if (value > max) {
            return max;
        } else {
            return value;
        }
    }
}

File: occasum-core/src/main/kotlin/pw/stamina/occasum/properties/primitives/clamp/MaxDoubleClamp.kt
package pw.stamina.occasum.properties.primitives.clamp

internal class MaxDoubleClamp(private val max: Double) : DoubleClamp {

    override fun clamp(value: Double): Double {
        return if (value > max) {
            max
        } else {
            value
        }
    }
}
